sql/core/src/test/resources/sql-tests/inputs/cte.sql

```diff
@@ -28,6 +28,177 @@ FROM   CTE1 t1
 WITH t(x) AS (SELECT 1)
 SELECT * FROM t WHERE x = 1;
 
+-- CTE in CTE definition
+WITH t as (
+  WITH t2 AS (SELECT 1)
+  SELECT * FROM t2
+)
+SELECT * FROM t;
+
+-- CTE in subquery
+SELECT max(c) FROM (
+  WITH t(c) AS (SELECT 1)
+  SELECT * FROM t
+);
+
+-- CTE in subquery expression
+SELECT (
+  WITH t AS (SELECT 1)
+  SELECT * FROM t
+);
+
+-- CTE in CTE definition shadows outer (legacy substitution)
+SET spark.sql.legacy.cte.substitution.enabled=true;
```

Comment:
Could we split these tests into two separate files: the legacy one and non-legacy one?

Reply:
ok, done